From a071bd2ed7631e6d759c758e3cc0a1c2cef538bc Mon Sep 17 00:00:00 2001
Date: Wed, 9 Nov 2016 18:49:25 +0100
Subject: [PATCH] In SafelyProcessEvent(), on an exception in the underlying
 ProcessEvent(), signal that processing of the current event was finished.
 (Otherwise it would be given to the next event handler in the list, i.e. be
 processed twice.)

---
 src/common/event.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/common/event.cpp b/src/common/event.cpp
index 2faf299..f519dd8 100644
--- a/src/common/event.cpp
+++ b/src/common/event.cpp
@@ -1664,7 +1664,10 @@ bool wxEvtHandler::SafelyProcessEvent(wxEvent& event)
         }
     }
 
-    return false;
+
+    // Signal that processing of the current event was finished.
+    // (Otherwise it would be given to the next event handler in the list, i.e. be processed twice.)
+    return true;
 #endif // wxUSE_EXCEPTIONS
 }
 
-- 
2.10.0.windows.1

